### PR TITLE
tweaking rust-libp2p permissions

### PR DESCRIPTION
**Pull request by dhuseby**
### Summary
This permissions change does a few things:
* It moves all of the "Repos - rust" team members to just members so that maintenance of that team has to go through an Admin of the libp2p org.
* It separates the rust-libp2p Maintainers into two teams: "rust-libp2p Active Maintainers" with just the currently active maintainers and "rust-libp2p Maintainers" with past and present maintainers that we still trust as reviewers. The former is granted Admin permissions over just the rust-libp2p repos. This was necessary because the other teams with Admin permissions have scopes much larger than the rust-libp2p repos.
* Correctly sets the "rust-libp2p Active Maintainers" to include @jxs as a member but NOT a maintainer of that team. He will have Admin rights over the libp2p repos but will not be able to add any members to the "rust-libp2p Active Maintainers" list without going through an Admin of the libp2p org.
* Correctly demotes @mxinden from a "maintainer" of the "rust-libp2p Maintainers" team down to just a "member"

### Why do you need this?
There are some cases where a trusted and active maintainer of the rust-libp2p repos must merge a PR and bypass the required checks. This is rare, but does happen.

### Reviewer's Checklist

- [ ] It is clear where the request is coming from (if unsure, ask)
- [ ] All the automated checks passed
- [ ] The YAML changes reflect the summary of the request
- [ ] The Terraform plan posted as a comment reflects the summary of the request

### CODE REVIEW

**Review by p-shahi — APPROVED** [review]

**Comment by galargh** [comment]
Hi! @laurentsenta pinged me about this PR. The issue is that `The Terraform plan posted as a comment reflects the summary of the request` is not satisfied. There must be some weird combination of team related modifications that we haven't hit before that causes terraform to want to replace all the team-member and team-repository relationships. We'll investigate more next week and will come up with a solution.

Thank you @andyschwab for proceeding with the revert (https://github.com/libp2p/github-mgmt/pull/215). It has been successfully applied (https://github.com/libp2p/github-mgmt/actions/runs/8303217362) which means we should be back to where we started.

I'm going to run the Plan workflow from master now to see if it's clean. Then, we could try to introduce the changes from this PR in smaller chunks.

**Comment by galargh** [comment]
The plan on master is clean - https://github.com/libp2p/github-mgmt/actions/runs/8303310226/job/22727225659

It looks like the process of creating a new team via github-mgmt is currently broken - https://github.com/libp2p/github-mgmt/pull/216
Modifying an existing team works just fine - https://github.com/libp2p/github-mgmt/pull/217

To unblock you, I created the `rust-libp2p Active Maintainers` manually and ran the Sync workflow - https://github.com/libp2p/github-mgmt/actions/runs/8303362271 Once it's complete, you can proceed with other changes.

**Comment by galargh** [comment]
The sync is complete ✅ You can now proceed with the other changes. We'll investigate in detail why creation (and, I suspect, deletion) of teams causes these huge plans to be created now, and we'll update github-mgmt with a fix as soon as possible.

**Review by mxinden — COMMENTED** [review]
There is an open security vulnerability in rust-yamux:

https://github.com/libp2p/rust-yamux/security/advisories/GHSA-3999-5ffv-wp2r

I am helping with a proposed fix:

https://github.com/libp2p/rust-yamux-ghsa-3999-5ffv-wp2r/pull/1

I lost access to the above pull request sometime within the last couple of days. I assume it is due to this pull request.

I suggest either reverting this pull request or giving me access to the above rust-yamux pull request?

//CC @jxs

**Comment by dhuseby** [comment]
@mxinden My apologies for the inconvenience. There was a bug in the teraform action that caused all of the teams to be deleted in response to creating a new team. @galargh manually fixed it and we've since updated the permissions. Everything should be back to normal now.

**Pull request merged**
